sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs

```diff
@@ -1946,29 +1941,43 @@ public partial class RangeFacetResult<T> where T : struct
         public T? From { get { throw null; } }
         public T? To { get { throw null; } }
     }
-    public partial class SearchDocument : System.Dynamic.DynamicObject, System.Collections.Generic.ICollection<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.Generic.IDictionary<string, object>, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.IEnumerable
+    public partial class SearchDocument : System.Collections.Generic.ICollection<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.Generic.IDictionary<string, object>, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.IEnumerable
     {
         public SearchDocument() { }
         public SearchDocument(System.Collections.Generic.IDictionary<string, object> values) { }
         public int Count { get { throw null; } }
         public object this[string key] { get { throw null; } set { } }
         public System.Collections.Generic.ICollection<string> Keys { get { throw null; } }
         bool System.Collections.Generic.ICollection<System.Collections.Generic.KeyValuePair<System.String,System.Object>>.IsReadOnly { get { throw null; } }
-        public System.Collections.Generic.ICollection<object> Values { get { throw null; } }
+        System.Collections.Generic.ICollection<object> System.Collections.Generic.IDictionary<System.String,System.Object>.Values { get { throw null; } }
         public void Add(string key, object value) { }
         public void Clear() { }
         public bool ContainsKey(string key) { throw null; }
+        public bool? GetBoolean(string key) { throw null; }
+        public bool[] GetBooleanArray(string key) { throw null; }
+        public System.DateTimeOffset? GetDateTimeOffset(string key) { throw null; }
+        public System.DateTimeOffset[] GetDateTimeOffsetArray(string key) { throw null; }
+        public double? GetDouble(string key) { throw null; }
+        public double[] GetDoubleArray(string key) { throw null; }
         public System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<string, object>> GetEnumerator() { throw null; }
+        public int? GetInt32(string key) { throw null; }
+        public int[] GetInt32Array(string key) { throw null; }
+        public long? GetInt64(string key) { throw null; }
+        public long[] GetInt64Array(string key) { throw null; }
+        public Azure.Search.Documents.Models.SearchDocument GetObject(string key) { throw null; }
+        public Azure.Search.Documents.Models.SearchDocument[] GetObjectArray(string key) { throw null; }
+        public Azure.Core.Spatial.PointGeometry GetPoint(string key) { throw null; }
```

Comment:
Only GetPoint? what about other geometries?

Reply:
[Search only supports Point in documents and Polygon in queries.](https://docs.microsoft.com/en-us/rest/api/searchservice/supported-data-types)

Reply:
Sorry I’ve been out of the loop, but why `PointGeometry`? In EDM it’s `GeographyPoint` which is distinct from `GeometryPoint`.

Reply:
So we should change it to [GeometryPosition?](https://apiview.dev/Assemblies/Review/8f2e023e377f4627a4ec815f6e11d013#Azure.Core.Spatial.GeometryPosition)

```C#
public readonly struct GeometryPosition : IEquatable<GeometryPosition> {
    public GeometryPosition(double longitude, double latitude);
    public GeometryPosition(double longitude, double latitude, double? altitude);
    public double? Altitude { get; }
    public double Latitude { get; }
    public double Longitude { get; }
    public static bool operator ==(GeometryPosition left, GeometryPosition right);
    public static bool operator !=(GeometryPosition left, GeometryPosition right);
    public bool Equals(GeometryPosition other);
    public override bool Equals(object? obj);
    public override int GetHashCode();
    public override string ToString();
}
```

Reply:
>GeometryPosition?

No, I don't think so.

>Sorry I’ve been out of the loop, but why PointGeometry? In EDM it’s GeographyPoint which is distinct from GeometryPoint.

We decided not to make a distinction in Azure.Core implementation of Spatial types. @KrzysztofCwalina had opinions about the type names that match the spec exactly so they were adjusted a bit.

See the API review at https://apiview.dev/Assemblies/Review/8f2e023e377f4627a4ec815f6e11d013#Azure.Core.Spatial

Reply:
I'll leave it `PointGeometry` for now, but I'm willing to do the Ask Mode work for any breaks here if our mind change as `Azure.Core.Experimental` iterates.

Reply:
These are different things though. In OData, `GeographyPoint` assumes a round-Earth co-ordinate system, but `GeometryPoint` uses a flat co-ordinate system. While it's unlikely Search will ever support the flat system, what if another service does in the future?

Reply:
We can support it by having a coordinate system reference as part of geometry properties as an option.

Currently we don't provide any operation on geometries in the Azure.Core itself so for us these types are DTOs only.

Cosmos does support flat coordinate system but it's a fully service-side setting that only affects indexing/querying.